src/lib/alloc.c

```diff
@@ -92,6 +93,7 @@ static inline uint32_t heap_get_size(struct mm_heap *heap)
 
 	return size;
 }
+#endif
```

Comment:
Can you remove L75-96? I assume clang complains that they are not used. It seems these have been dead code for long, not used for anything, so can be removed. Please add a separate commit.